VTUL/iawa
# LIBTD-1257: IAWA migration from Hyrax 2.0.0 based on Hyrax 2.1.0

**JIRA Ticket**: https://webapps.es.vt.edu/jira/browse/LIBTD-1257

# What does this Pull Request do?
This PR modifies the IAWA application from Hyrax gem 2.0.0 to 2.1.0.

# What are the changes?

Modifications include the following:
* Modifications following the Hyrax 2.1.0 Release Notes (https://github.com/samvera/hyrax/releases/tag/v2.1.0). 
 * Changes affecting Collections that include the presenter, form, and multiple views.
 * A change to use the default Hyrax 2.1 user_util_links for easy access to the user dashboard. 
* A temporary solution for manually setting the CollectionType constants. This will need to be updated once the following Ticket is resolved: https://webapps.es.vt.edu/jira/browse/LIBTD-1419
 * A temporarily fix to enable static files from public folder to ensure that profile images and collection branding images are accessible in production. This is related to an issue similarly documented here: https://webapps.es.vt.edu/jira/browse/LIBTD-1423

# How should this be tested?

To test this, please use the InstallScripts branch `LIBTD-1257`. This will ensure that the appropriate default Hyrax CollectionTypes are installed. Also, use the following settings in your `site_secrets.yml` :

```
project_name: 'iawa'
project_git_identifier: 'LIBTD-1257-dev'
project_app_env: 'production'
```

Note that the `project_secret_key_base` and `project_cas_url` will also need to be updated.

Try the following:
* Attempt to create a collection, both with and without branding images.
* Attempt to create one or more works, both with and without images. 
* Attempt to associate a work with a collection.
* Attempt to modify your user profile avatar.
* Attempt to access the dashboard from the upper right drop down.
* Try anything else and note any irregularities. 

# Additional Notes:
* This branch was created from LIBTD-1256, picking up from a Hyrax 2.0.0 version of IAWA.
* This PR does not address any existing import scripts from the Hyrax 1.0 version, which may be affected with this change when it is eventually merged into master.
* This PR needs to be tested in production mode due to an open issue with the development mode and the Mirador viewer: https://webapps.es.vt.edu/jira/browse/LIBTD-1343
* Some changes (like Collection and Presenter overrides) were taken and modified from https://github.com/VTUL/hyrax2
* Some notes of the migration were documented on the wiki: https://github.com/VTUL/iawa/wiki/IAWA-Migration-Notes:-Hyrax-v.-2.0.0-to-v.-2.1.0

# Interested parties
@tingtingjh
